### PR TITLE
Materials.json - corrected typos and reordered entries

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Infrastructure "Alphabetically sorts materials in materials.json and corrects typos"```

#### Purpose of change
I was looking through matterials.json, and noticed that while most of it was alphabetically sorted, the latter part (newer additions) was not. I also saw some spelling errors in the names of some materials.

#### Describe the solution
- Fixed the names of the materials. I capitalised the second word in some material names (Processed Food, Ice Cream), to fit with earlier material names
- Fixed typos in the names of (formerly) Processed Fod and Dried Vegetable to Processed Food and Dried Vegetables
- Alphabetically sorted all the materials. I sorted them by material id name, so Biosilicified Chitin (acidchitin is the ID) comes before Alcohol (alcohol is the ID)

#### Additional context
I couldn't see anything that would break by sorting the materials/fixing their names, but let me know if there is any issues.

### CODE REVIEW

File: data/json/materials.json [hunk 1]
```diff
@@ -45,25 +68,28 @@
...
-    "cut_resist": 4,
-    "acid_resist": 13,
-    "fire_resist": 2,
-    "elec_resist": 2,
-    "chip_resist": 10,
-    "dmg_adj": [ "scratched", "cut", "cracked", "shattered" ],
-    "bash_dmg_verb": "cracked",
-    "cut_dmg_verb": "chipped",
+    "ident": "bean",
+    "name": "Bean",
```
Comment (Night-Pryanik) [inline data/json/materials.json]: `Bean` should be earlier than `Biosilicified Chitin`.

Comment (SourSlime) [reply data/json/materials.json]: I ordered them by ID, not name (see the pull request solution section for more details)

Comment (Night-Pryanik) [reply data/json/materials.json]: Well, in that case it's ok. Though I'd personally prefer sorting by their name.

**Comment by kevingranade** [comment]
It won't break anything per se, but it makes noise in commit histories and
such and can cause merge conflicts. We'd rather people not reorder things.

**Comment by SourSlime** [comment]
Thanks for letting me know, I'll keep that in mind for next time.

**Comment by matthemsteger** [comment]
Hmmm, looking at this brings up another issue potentially. It looks like all the burn data has an implicit time component that wasnt changed with the new turn length?

File: data/json/materials.json (hunk 1 above)
Comment (SourSlime) [reply data/json/materials.json]: I agree, and I did that initially, but then I noticed that other files like armor.json also sort by item ID

**Comment by SourSlime** [comment]
If that’s true, I could fix that in this PR

**Comment by matthemsteger** [comment]
Did not mean to derail .. should probably be a different PR. I suspect it requires code changes too.

**Comment by mlangsdorf** [comment]
it's probably beyond the scope of this PR, but redoing density in kg/m^3 instead of arbitrary unitless numbers would be useful.


Material | Current Density | Real Value (kg/m^3)
== | == | ==
Aluminum | 10 | 2700
Budget Steel | 30 | 7850
Copper | 59 | 8960
Gold | 130 | 19320
Hard Steel | 40 | 7750
Iron | 52 | 7870
Lead | 100 | 11360
Platinum | 140 | 21450
Silver | 70 | 10490
Steel | 30 | 7750

It doesn't really matter, because we don't use material::density() except in collisions, but I'd still rather have sensible values.

**Comment by kevingranade** [comment]
Please back out the reordering changes, they are causing this to conflict.

**Comment by SourSlime** [comment]
Ok. I'm going to open a new PR with the typo fixes

**Pull request closed without merging**